data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template

```diff
@@ -13,19 +13,21 @@ set -euo pipefail
 
 RELEASE_IMAGE={{.ReleaseImage}}
 
-echo "Pulling $RELEASE_IMAGE..."
-while true
-do
-    record_service_stage_start "pull-release-image"
-    if podman pull --quiet "$RELEASE_IMAGE"
-    then
-        record_service_stage_success
-        break
-    else
-        record_service_stage_failure
-        echo "Pull failed. Retrying $RELEASE_IMAGE..."
-    fi
-done
+image_found=$(podman images --format json | jq --arg releaseimage "$RELEASE_IMAGE" 'any(.[].Names; .[] == $releaseimage)')
+if [[ "$image_found" == "false" ]]; then
+   echo "Pulling $RELEASE_IMAGE..."
+   while true; do
+       record_service_stage_start "pull-release-image"
+       if podman pull --quiet "$RELEASE_IMAGE"
+       then
+           record_service_stage_success
```

Comment:
We probably want to still `record_service_stage_success` even if we don't need to pull an image.